src/util/tools.h

```diff
@@ -234,19 +234,22 @@ class EnumIterator {
 };
 
 /// \brief Get value of map, iff not key is not in map return defval
-template <typename K, typename V>
-V getValueOrDefault(const std::vector<std::pair<K, V>>& m, const K& key, const V& defval)
+template <typename K, typename V, typename C>
+V getValueOrDefault(const C& m, const K& key, const V& defval)
 {
+    if constexpr (std::is_same_v<C, std::map<K, V>>) {
+        auto it = m.find(key);
+        if (it == m.end())
+            return defval;
+        return it->second;
+    }
+
     auto it = std::find_if(m.begin(), m.end(), [&](auto&& v) { return v.first == key; });
-    return (it == m.end()) ? defval : it->second;
+    if (it == m.end())
+        return defval;
+    return it->second;
 }
 
-template <typename K, typename V>
-V getValueOrDefault(const std::map<K, V>& m, const K& key, const V& defval)
-{
-    auto it = m.find(key);
-    return (it == m.end()) ? defval : it->second;
-}
```

Comment:
Does it work also, if you leave this? I don't like the `constexpr` - it needs explanation like 
```
// if container is map, use map methods instead
```

Reply:
if constexpr is specifically made to work with templates.

Reply:
I tried to make it more generic (if find() method exists, use that). Couldn't find a simple way to do that. On that note, it seems only std::map has one.